### ISSUE TITLE
Accurate collision checking functions.

### ISSUE THREAD

**Issue by ianthe-one** [YoYoGames/GameMaker-Bugs#4358]
### Is your feature request related to a problem?

Using collision functions like `place_meeting`, `collision_rectangle[_list]`, `instance_place`, etc... in recent version don't check for overlap correctly, as the documentation describes [here](https://manual.gamemaker.io/monthly/en/GameMaker_Language/GML_Reference/Movement_And_Collisions/Collisions/place_meeting.htm):
![image](https://github.com/YoYoGames/GameMaker-Bugs/assets/118121781/3e6bb0e7-e8b2-4245-bb1a-56cfb50f9bab)  
&nbsp;
And [here](https://manual.gamemaker.io/monthly/en/GameMaker_Language/GML_Reference/Movement_And_Collisions/Collisions/Collisions.htm):
![image](https://github.com/YoYoGames/GameMaker-Bugs/assets/118121781/07a6188a-b5dd-4e92-96ec-974563c93e6e)  
&nbsp;
&nbsp;
The coordinates of the objects being checked with/against get rounded, which causes issues if you have a system that uses non-integer velocities/positions. said issues don't occur when using proper AABB overlap checking, like with the function `rectangle_in_rectangle`. Checking 'Collision Compatibility Mode' or not, doesn't change anything, just rounds the `bbox_*` values.  
&nbsp;
&nbsp;
Demonstration:

https://github.com/YoYoGames/GameMaker-Bugs/assets/118121781/7cb71018-7a58-46e3-9fec-9a5bb918e3a8

&nbsp;
&nbsp;
It doesn't even make sense that it works like this in the first place, especially for sprites with a `Rectangle` or `Rectangle With Rotation` collision masks. And it's made me all but completely abandon GameMaker entirely because it was nearly impossible to get smooth sub-pixel movement and proper collisions working.
&nbsp;
*******
&nbsp;

### Describe the solution you'd like

Disabling the rounding, or fixing the functions would be enough really. Maybe the `Collision Compatibility Mode` could re-enable the current functionality instead of practically doing nothing.
&nbsp;
*******
&nbsp;

### Describe alternatives you've considered

There is a workaround, but it's expensive, generally a pain to write, and **it only works with middle-centered AABBs**.
```gml
#macro padding 4

function place_meeting_fix(x, y, obj) {
    var hw = (self.bbox_right - self.bbox_left) / 2,
        hh = (self.bbox_bottom - self.bbox_top) / 2,
        list = ds_list_create();
    
    collision_rectangle_list(
        x - hh - padding, y - hw - padding,
        x + hh + padding, y + hw + padding,
        obj, true, true, list, true
    );

    var list_size = ds_list_size(list);
    for(var i = 0; i < list_size; i++) {
        var c_obj = ds_list_get(list, i),
            collided = rectangle_in_rectangle(
                self.bbox_left, self.bbox_top,
                self.bbox_right, self.bbox_bottom,
                c_obj.bbox_left, c_obj.bbox_top,
                c_obj.bbox_right, c_obj.bbox_bottom
            );

        if(collided) return true;
    }

    ds_list_destroy(list);
    return false;
}
```
&nbsp;
&nbsp;
You could write a proper system to be able to work with other center-points or different shapes, but:
1. It would get REALLY expensive;
2. It would be really complex;
3. It should be built-in functionality.

Speaking from experience.
&nbsp;
*******
&nbsp;
### Additional context

This is a feature request and not a bug report because the documentation says it's working as intended.

**Comment by gnysek** [comment]
I suspect that first quote form manual is out of date now, so this might be not a feature request, but bug in documentation, however smarter people should check this.

**Comment by ianthe-one** [comment]
> I suspect that first quote form manual is out of date now, so this might be not a feature request, but bug in documentation, however smarter people should check this.

Could be. Either way I'd like proper collision detection

**Comment by BackYard321** [comment]
> I suspect that first quote form manual is out of date now, so this might be not a feature request, but bug in documentation, however smarter people should check this.

gnysek is correct - when I checked, `collision compatibility` not only rounds bboxes, but also performs the collision checking rounding. If it's not enabled, it uses floats (ie what you're talking about). So the problem you're experiencing stems purely from the "middle of pixel" thing.

I definitely feel your frustration re: middle of pixel VS outer edges, and two years ago I was considering dropping GM for the same reason, since I'm also working on a platformer that requires pixel-precision. But, the more I delved into it, the more I started seeing how beneficial it is. It provides a degree of wiggle room, which ultimately made my code cleaner and less buggy!
